src/test/java/org/jabref/logic/importer/fetcher/GrobidCitationFetcherTest.java

```diff
@@ -0,0 +1,206 @@
+     */
+    @Test
+    public void grobidParseBibToBibEntryTest() throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, example1AsBibtex);
+        assertTrue(bibEntry.isPresent());
+        assertEquals(bibEntry.get().getField(StandardField.AUTHOR).get(), "T Derwing and M Rossiter and M Munro");
+        assertEquals(bibEntry.get().getField(StandardField.TITLE).get(), "Teaching native speakers to listen to foreign-accented speech");
+        assertEquals(bibEntry.get().getField(StandardField.JOURNAL).get(), "Journal of Multilingual and Multicultural Development");
+        assertEquals(bibEntry.get().getField(StandardField.YEAR).get(), "2002");
+    }
+
+    /**
+     * Checks if parseBibToBibEntry creates a empty BibEntry on empty response
+     */
+    @Test
+    public void grobidParseBibToBibEntryReturnsEmptyOptionalWhenFailedTest() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, defaultReturnValue);
+        assertTrue(bibEntry.isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.AUTHOR).isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.TITLE).isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.YEAR).isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.JOURNAL).isPresent());
+    }
+
+    /**
+     * When the user runs the program without passing anything, no BibEntry is generated (The Optional should be empty)
+     */
+    @Test
+    public void grobidParseBibToBibEntryReturnsNoBibEntryWhenPassingInvalidInputTest() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, "");
+        assertFalse(bibEntry.isPresent());
+    }
+
+
+    /**
+     * Tests if the performSearch function correctly splits the plain reference text into the right parts.
+     */
+    @Test
+    public void grobidPerformSearchCorrectlySplitsStringTest() throws FetcherException {
+        List<BibEntry> entries = grobidCitationFetcher.performSearch(example3 + ";;" + example4);
+        assertTrue(entries.size() == 2);
```

Comment:
Use "assertEquals(List.of(bibEntryExample3, bibEntryExample4), entries)". Hope, this works. Otherwise, a conversion to a set and comparing that could work.